### PR TITLE
Adding rubocop

### PR DESCRIPTION
**Pull request by JelF**
Two files: rubocop.yml and rubocop.rails.yml to handle both cases

### CODE REVIEW

File: rubocop/lib/rubocop.yml
Comment (JelF) [inline rubocop/lib/rubocop.yml]: Following lines are exactly match default values in rubocop 0.48.1 (magical coyp-and-paste method). They would be removed before merge

File: rubocop/.gitignore
Comment (akxcv) [inline rubocop/.gitignore:3]: Why would you ignore `Gemfile.lock`?

File: rubocop/.rubocop.yml
Comment (akxcv) [inline rubocop/.rubocop.yml]: What does `eslint-config-umbrellio` have to do with Rubocop?

File: rubocop/README.md
Comment (akxcv) [inline rubocop/README.md]: "Umbrellio". Please don't get the spelling wrong when it's crucial :)

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: I thought that was the default (`if foo = 123` is disallowed). Am I wrong?

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: Hmm. In my opinion, this is a good cop to keep enabled. It disallows "magical" conditions like `if 20` or `if true` and forces you to write more self-documenting code.

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: Why?

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: 👍

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: I think we should decide on a single enforced style.

File: rubocop/lib/rubocop.yml
Comment (akxcv) [inline rubocop/lib/rubocop.yml]: There should be no spaces allowed between `[` and `]`, IMHO.

File: rubocop/rubocop.gemspec
Comment (akxcv) [inline rubocop/rubocop.gemspec]: Again, this gem should have a different name.

File: rubocop/rubocop.gemspec
Comment (akxcv) [inline rubocop/rubocop.gemspec]: Whoa. I suggest you force push different credentials.

File: rubocop/rubocop.gemspec
Comment (akxcv) [inline rubocop/rubocop.gemspec]: Again, typos in crucial places.

**Review by akxcv — CHANGES_REQUESTED** [review]
Please, ask someone else that writes a lot of Ruby code to review this PR. I'm afraid I'm not the most qualified person to accept such things :)

File: rubocop/.gitignore
Comment (JelF) [reply rubocop/.gitignore:3]: It is a standard behaviour for all gems. gemspec should strictly describe with wich gems it works.
In this case, both runtime dependencies are locked to exact versions, so there is no real reason to ignore it, but it is still a common behaviour so i don't think we need to change it

File: rubocop/.rubocop.yml
Comment (JelF) [reply rubocop/.rubocop.yml]: Most reason is Rakefile, which can possibly by used for `.yml.erb`, if we will need it. I also use it for gemspec. And, finnaly, it allows to run `$ bundle exec rubocop --show-cop`

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: You are exactly right, do i need to reword comment?

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: I'll move it into rubocop.rails.yml and disable it for .js.rb there

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: I personaly prefer `if..elsif..else..end` and community seems to prefer it too. There is only style purposes

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: mixed means that we have to write `/foo/` but `%r{foo/bar}`. As far as i see, we already do it

File: rubocop/rubocop.gemspec
Comment (JelF) [reply rubocop/rubocop.gemspec]: lol

File: rubocop/rubocop.gemspec
Comment (JelF) [reply rubocop/rubocop.gemspec]: oops

File: rubocop/.rubocop.yml
Comment (JelF) [reply rubocop/.rubocop.yml]: I understood what was the mistake :)

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: I did some invertigation, most violations was like `/http:\/\/...`

File: rubocop/lib/rubocop.yml
Comment (akxcv) [reply rubocop/lib/rubocop.yml]: Yes, because your comment seems to suggest the opposite :)

File: rubocop/lib/rubocop.yml
Comment (akxcv) [reply rubocop/lib/rubocop.yml]: Why? And also, in my opinion, `.js.rb` files should not be ignored in this gem. The project that uses the gem is responsible for this.

File: rubocop/lib/rubocop.yml
Comment (akxcv) [reply rubocop/lib/rubocop.yml]: Yes, I agree, maybe I mistakenly thought you were suggesting the opposite.

File: rubocop/lib/rubocop.yml
Comment (akxcv) [reply rubocop/lib/rubocop.yml]: Yes, I know that we use both styles. But I don't think that's good. And you don't, either, judging by your use of the word "violation" :)

**Comment by akxcv** [comment]
Also, AFAIK, there are possible issues with making this a gem. Please discuss the details with @fiscal-cliff

**Comment by akxcv** [comment]
Also also: maybe we should split the big `.rubocop.yml` file into multiple smaller files, like `.rubocop.lint.yml`, `.rubocop.style.yml` and so on. I think I should do the same over in the ESLint config.

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: Agree

**Comment by JelF** [comment]
> Also, AFAIK, there are possible issues with making this a gem. Please discuss the details with @fiscal-cliff

i already tested it as a gem, it works

> Also also: maybe we should split the big .rubocop.yml file into multiple smaller files, like .rubocop.lint.yml,.rubocop.style.yml` and so on. I think I should do the same over in the ESLint config.

Reasonable, but can be confusing in some cases. I don't want to do it unless this gem would be proved to be stable

File: rubocop/lib/rubocop.yml
Comment (tycooon) [reply rubocop/lib/rubocop.yml]: Personally I prefer `if foo = 123`.

File: rubocop/lib/rubocop.yml
Comment (tycooon) [inline rubocop/lib/rubocop.yml]: OK

File: rubocop/lib/rubocop.yml
Comment (tycooon) [inline rubocop/lib/rubocop.yml]: I remember having problems with it, but let’s try.

File: rubocop/lib/rubocop.yml
Comment (tycooon) [reply rubocop/lib/rubocop.yml]: I prefer blank `case` because it’s more clear that all conditions are on the same level (`if` and `elsif` have different length).

File: rubocop/lib/rubocop.yml
Comment (tycooon) [inline rubocop/lib/rubocop.yml]: Some ruby regexp-methods (like `scan`) make it hard to properly get matches so we use variables like `$1` there. I think we can enable this cop and just disable it locally where needed.

File: rubocop/lib/rubocop.yml
Comment (tycooon) [reply rubocop/lib/rubocop.yml]: > mixed: Use slashes on single-line regexes, and `%r` on multi-line regexes.
I think `mixed` is ok.

File: rubocop/lib/rubocop.yml
Comment (tycooon) [inline rubocop/lib/rubocop.yml]: 👍

File: rubocop/lib/rubocop.yml
Comment (tycooon) [reply rubocop/lib/rubocop.yml]: Because i find it more consistent: 
`[ { foo: 123 } ]`
`{ { foo: 123 } }`

**Comment by JelF** [comment]
I think it sets up all cops we already use, let's integrate it?

**Review by tycooon — APPROVED** [review]

File: rubocop/README.md
Comment (akxcv) [inline rubocop/README.md]: What's this for? Seems dangerous and unnecessary to me.

File: rubocop/rubocop.gemspec
Comment (akxcv) [inline rubocop/rubocop.gemspec]: Why? Are you trying to sync code style version to rubocop version? What for?

File: rubocop/README.md
Comment (JelF) [reply rubocop/README.md]: Just in case. Maybe i should reword 'Optional'?

File: rubocop/lib/rubocop.yml
Comment (fiscal-cliff) [inline rubocop/lib/rubocop.yml:20]: Why don't you enable the cop?

File: rubocop/rubocop.gemspec
Comment (JelF) [reply rubocop/rubocop.gemspec]: Semantic versioning does not make sence here, i used model which is common when a gem requires exact version of another. If this will be improved by adding another version number, like '0.48.1.0', would it be ok?

File: rubocop/lib/rubocop.rails.yml
Comment (JelF) [inline rubocop/lib/rubocop.rails.yml]: TODO: check for unnecessary blank lines

File: rubocop/lib/rubocop.yml
Comment (fiscal-cliff) [inline rubocop/lib/rubocop.yml]: The same question here

File: rubocop/rubocop.gemspec
Comment (fiscal-cliff) [reply rubocop/rubocop.gemspec]: Looks reasonable

File: rubocop/.gitignore
Comment (fiscal-cliff) [reply rubocop/.gitignore:3]: Gemfile.lock is ignored by default. The same story [here](https://github.com/fiscal-cliff/sequel-tablefunc/blob/master/.gitignore) for instance

File: rubocop/README.md
Comment (fiscal-cliff) [reply rubocop/README.md]: You had better replace the work "Optional" with the word "OR".

File: rubocop/README.md
Comment (fiscal-cliff) [inline rubocop/README.md]: Sounds wacky

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml:20]:
```
def foo
  foo!
rescue SpecificFooException
end
```

File: rubocop/lib/rubocop.yml
Comment (JelF) [reply rubocop/lib/rubocop.yml]: I will enable this cop, thank you

File: rubocop/lib/rubocop.yml
Comment (fiscal-cliff) [reply rubocop/lib/rubocop.yml:20]: so what?! Looks like you ignore exception which is not ok in general. It is better to disable the cop in special cases, and enforce by default

File: rubocop/README.md
Comment (fiscal-cliff) [inline rubocop/README.md]: omit ", branch: 'master'`"

**Review by fiscal-cliff — APPROVED** [review]
Just a few recomendations left. I approved. Go ahead!

**Review by akxcv — APPROVED** [review]

**Pull request merged**
